scripts/zkai_snarkjs_statement_envelope_benchmark.py

```
@@ -0,0 +1,523 @@
+def _check_statement_policy(statement: dict[str, Any]) -> None:
+    for key, expected in EXPECTED_STATEMENT.items():
+        if statement.get(key) != expected:
+            raise SnarkjsEnvelopeError(f"statement policy mismatch for {key}")
+    if statement.get("schema") != STATEMENT_SCHEMA:
+        raise SnarkjsEnvelopeError("unsupported statement schema")
+
+
+def _check_artifact_hashes(envelope: dict[str, Any]) -> pathlib.Path:
+    artifacts = envelope.get("artifacts")
+    if not isinstance(artifacts, dict):
+        raise SnarkjsEnvelopeError("artifacts must be an object")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)